README.md

```diff
@@ -1,34 +1,148 @@
+%}
+```
+
+#### Delete 
+
+You can also delete an existing embedding by invoking `modules/openai/embeddings/delete` command:
+
+```
+{% function pos_embedding = 'modules/openai/commands/embeddings/delete', id: embedding.id %}
+```
+
+#### Read
+
+To leverage embeddings, at some point you would need to perform search operation through `modules/openai/queries/embeddings/search.liquid`. Usually you would like to sort the results by relevancy to the embedding that is provided as an input (for example, user enters their search query, which you transform into an embedding using `fetch_embedding` command, and you want to get 5 most relevant embeddings):
+
+```
+  function related_embeddings = 'modules/openai/queries/embeddings/search', related_to: embedding, limit: 5
 ```
 
+## Example application
+
+One of the typical use case of using Embedding is to implement AI enhanced search. This is also the first step of RAG (Retrieval-Augmented Generation). You can find an example implementation of such functionality in the `app` directory, ready to be copied and customized for your application needs, or further integration with LLM of your choice. The example shows example implementation of key concepts:
+
+### A script to transform existing data into Embeddings
+
+You can find the script in `app/lib/commands/openai/commands_to_embeddings.liquid`. It fetches the existing [Pages](https://documentation.platformos.com/developer-guide/pages/pages), filters them by your needs, and creates embeddings based on their content. This is meant for demonstration purposes - instead of creating embeddings for the whole page, you might want for example to create embeddings for certain paragraphs, which probably would yield a better results.
+
+The script can be run multiple times and will remove embeddings of pages that have been removed since the last run, will update the pages that have changed, and finally will add new embeddings for new pages.
+
+the script assumes that each Page that is index has configured [Metadata](https://documentation.platformos.com/developer-guide/pages/metadata) - `title` and `description`. Those will be required to efficiently render [Search Page](#search-page) without having to make another query to the DB.
+
+### Search page
+
+The example search page is located at `app/views/pages/openai_search.liquid`. It is just for demonstration purposes, and it violates some of the [best practices](https://documentation.platformos.com/developer-guide/modules/platformos-modules#how-to-use-devkit) like separating business rules from the presentation layer. This is intentional, as the page should just serve as an all-in-one example of the whole flow. 
+
+You will find there additional checks if the module [setup](#setup) has been done correctly, it supports hcaptcha challenge to avoid spam requests and demonstrates how to use the [commands and queries](#functionality-provided-by-the-user-module) provided by the OpenAI module.
```

Comment:
```suggestion
You'll find there additional checks to ensure the module [setup](#setup) has been configured correctly. It supports the hCaptcha challenge to prevent spam requests and demonstrates how to use the [commands and queries](#functionality-provided-by-the-user-module) provided by the OpenAI module.
```